tests/pytests/test_coordinator.py

```diff
@@ -121,3 +121,41 @@ def test_error_propagation_from_shards(env):
     #   1. The language requested in the command.
     #   2. The scorer requested in the command.
     #   3. Parameters evaluation
+
+def test_timeout(env):
+    """Tests that timeouts are handled properly by the coordinator.
+    We check that the coordinator returns a timeout error when the timeout is
+    reached in the shards or in the coordinator itself.
+    """
+
+    SkipOnNonCluster(env)
+    conn = getConnectionByEnv(env)
+
+    # Create the index
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'title', 'TEXT').ok()
+
+    # Populate the database with 1500 * nshards documents
+    n_docs = int(1500 * env.shardsCount)
+    for i in range(n_docs):
+        conn.execute_command('HSET', i ,'t1', str(i))
+
+    # Set the timeout policy to fail upon experiencing a timeout
+    env.expect('_FT.CONFIG', 'SET', 'ON_TIMEOUT', 'FAIL').ok()
```

Comment:
Does this set the policy for all shards or only the one we are connected to?

Reply:
Looks like only the shard. I'll use a new env with this configuration instead.